crates/ra_fmt2/src/dsl.rs

```diff
@@ -0,0 +1,198 @@
+use crate::pattern::{Pattern, PatternSet};
+use crate::trav_util::{next_non_whitespace_sibling, prev_non_whitespace_sibling};
+use ra_syntax::{SyntaxElement, SyntaxKind::*};
+
+/// `SpacingRule` describes whitespace requirements between `SyntaxElement` Note
+/// that it doesn't handle indentation (first whitespace on a line), there's
+/// `IndentRule` for that!
+#[derive(Debug, Clone)]
+pub(crate) struct SpacingRule {
```

Comment:
Those `pub(crate)` can normally be `pub` without `(crate)` when the current module is private.

Reply:
I *think* that would become an `unreachable_pub` warning in this edition